connectors/sync_job_runner.py

```diff
@@ -516,19 +523,45 @@ async def generator(self):
                 raise UnsupportedJobType
 
     async def update_ingestion_stats(self, interval):
+        last_checkpoint = None
+
         while True:
             await asyncio.sleep(interval)
 
             if not await self.reload_sync_job():
                 break
 
+            checkpoint = (
+                None
+                if not self.data_provider  # If we failed before initializing the data provider, we don't need to change the cursor
+                else (
+                    self.data_provider.checkpoint()
+                    if self.sync_job.is_content_sync()
+                    else None
+                )
+            )
+
+            if checkpoint != last_checkpoint:
+                self.sync_job.log_debug(
+                    "Connector reported a new checkpoint, triggering batch flush before saving"
+                )
+                last_checkpoint = checkpoint
+                # Fun stuff happens here - this will block until the data is flushed
+                # to Elasticsearch. Once it's flushed, we can continue and save the job state
+                # This is done to avoid data inconsistencies - if we don't flush, then
+                # we can store checkpoint before the data that preceded this checkpoint was ingested.
+                await self.sync_orchestrator.trigger_flush()
+                self.sync_job.log_debug("Data was successfully flushed before saving the checkpoint")
+
             result = self.sync_orchestrator.ingestion_stats()
             ingestion_stats = {
                 INDEXED_DOCUMENT_COUNT: result.get(INDEXED_DOCUMENT_COUNT, 0),
                 INDEXED_DOCUMENT_VOLUME: result.get(INDEXED_DOCUMENT_VOLUME, 0),
                 DELETED_DOCUMENT_COUNT: result.get(DELETED_DOCUMENT_COUNT, 0),
             }
-            await self.sync_job.update_metadata(ingestion_stats=ingestion_stats)
+            await self.sync_job.update_metadata(
+                ingestion_stats=ingestion_stats, checkpoint=checkpoint
+            )
```

Comment:
Can we add debug log that checkpoint was updated? Especially that we have 

`Data was successfully flushed before saving the checkpoint` above, so I would expect to see the logs "saving checkpoint" operation somewhere

Reply:
Added a log line in https://github.com/elastic/connectors/pull/2879/commits/366aba4296d74f1cf5a5ba22c4d9686fa107f0a4: I chose not to mention checkpoints cause logic here always updates the checkpoint anyway and sequence of log lines should be sufficient to understand what happened:

```
Connector reported a new checkpoint, triggering batch flush before saving
Data was successfully flushed before saving the checkpoint
Sync job metadata was updated successfully
```

Do you think we should still be more specific with this?